frontend/src/components/DataTable.tsx

```
@@ -0,0 +1,833 @@
+  useEffect(() => {
+    if (persistKey && columnFilters.length > 0) {
+      localStorage.setItem(`datatable_filters_${persistKey}`, JSON.stringify(columnFilters));
+    }
+  }, [columnFilters, persistKey]);
+
+  useEffect(() => {
+    if (persistKey) {
+      try {
+        const saved = localStorage.getItem(`datatable_visibility_${persistKey}`);
+        if (saved) setColumnVisibility(JSON.parse(saved));
+      } catch { }
```

ESLint

[error] Empty block statement.

(no-empty)